adapters/orbidder/orbidder.go

```diff
@@ -55,14 +36,34 @@ func (rcv *OrbidderAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *
 	headers.Add("Accept", "application/json")
 
 	return []*adapters.RequestData{{
-		Method:  "POST",
+		Method:  http.MethodPost,
 		Uri:     rcv.endpoint,
 		Body:    requestBodyJSON,
 		Headers: headers,
 	}}, errs
 }
 
-func preprocess(imp *openrtb2.Imp) error {
+// getValidImpressions validate imps and check for bid floor currency. Convert to EUR if necessary
+func getValidImpressions(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]openrtb2.Imp, []error) {
+	var errs []error
+	var validImps []openrtb2.Imp
+
+	for _, imp := range request.Imp {
+		if err := preprocessBidFloorCurrency(&imp, reqInfo); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		if err := preprocessExtensions(&imp); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		validImps = append(validImps, imp)
+	}
+	return validImps, errs
+}
+
+func preprocessExtensions(imp *openrtb2.Imp) error {
 	var bidderExt adapters.ExtImpBidder
 	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
```

Comment:
Are you just testing that unmarshalling `imp.ext` does not produce an error? Seems that you are throwing away the result.

Reply:
Yes, you are right. We only checking for errors in unmarshalling `imp.ext`. In the passt we had some problems with invalid Extensions send to our adapter, so we find this useful. We added soem testecases for the preprocessing.